crates/polars-sql/src/context.rs

```diff
@@ -704,38 +730,58 @@ impl SQLContext {
 
```

Comment:
L727 `let schema_before = lf.schema()?;`

Is problematic. It is very expensive on large dags as we do a complete conversion to IR to compute the schema. It also means we cannot resolve SQL server side.

Have to think of this a bit, but  maybe we should see if we can change our abstraction from `LazyFrame` to `IR` and build those.